sdk/resourcemanager/azure-resourcemanager-test/src/main/java/com/azure/resourcemanager/test/utils/TestUtilities.java

```diff
@@ -59,4 +70,80 @@ public static <T> int getSize(Iterable<T> iterable) {
     public static <T> boolean isEmpty(PagedIterable<T> iterable) {
         return !iterable.iterator().hasNext();
     }
+
+    /**
+     * Creates a comprehensive {@link TokenCredential} chain optimized for test environments.
+     * <p>
+     * This method constructs a credential chain that attempts multiple authentication methods
+     * in a specific order, making it suitable for various testing scenarios including local
+     * development and live tests in CI/CD pipelines.
+     * </p>
+     *
+     * <strong>Authentication Chain Order:</strong>
+     * <ol>
+     *   <li>{@link com.azure.identity.EnvironmentCredential} - Uses environment variables</li>
+     *   <li>{@link com.azure.identity.AzureCliCredential} - Uses Azure CLI authentication</li>
+     *   <li>{@link com.azure.identity.AzureDeveloperCliCredential} - Uses Azure Developer CLI</li>
+     *   <li>{@link com.azure.identity.AzurePipelinesCredential} - Uses Azure Pipelines service connection (if configured)</li>
+     *   <li>{@link com.azure.identity.AzurePowerShellCredential} - Uses Azure PowerShell authentication</li>
+     * </ol>
+     *
+     * <strong>Azure Pipelines Configuration:</strong><br>
+     * For Azure Pipelines authentication, the following environment variables must be set:
+     * <ul>
+     *   <li>{@code AZURESUBSCRIPTION_SERVICE_CONNECTION_ID}</li>
+     *   <li>{@code AZURESUBSCRIPTION_CLIENT_ID}</li>
+     *   <li>{@code AZURESUBSCRIPTION_TENANT_ID}</li>
+     *   <li>{@code SYSTEM_ACCESSTOKEN}</li>
+     * </ul>
+     *
+     * <strong>Local run Configuration:</strong><br>
+     * For local run authentication, use Azure CLI for example, the following environment variables must be set:
+     * <ul>
+     *   <li>{@code AZURE_SUBSCRIPTION_ID}</li>
+     *   <li>{@code AZURE_TENANT_ID}</li>
+     * </ul>
+     *
+     * @param isPlayback whether the test runs in playback mode.
+     *                          If in playback mode, returns a {@link MockTokenCredential}.
+     *                          Otherwise, returns the credential chain.
+     * @return a {@link TokenCredential} appropriate for the test environment:
+     *         {@link MockTokenCredential} for playback mode, or a
+     *         {@link com.azure.identity.ChainedTokenCredential} for live testing
+     * @see InterceptorManager#isPlaybackMode()
+     * @see MockTokenCredential
+     * @see com.azure.identity.ChainedTokenCredential
+     */
+    public static TokenCredential getTokenCredentialForTest(boolean isPlayback) {
```

Comment:
We should also support RECORD mode. So, instead of a `boolean` we can take `TestMode` as the input and we can support  PLAYBACK, RECORD and LIVE test modes.